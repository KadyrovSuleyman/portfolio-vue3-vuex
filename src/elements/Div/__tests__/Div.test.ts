import { mount, VueWrapper } from '@vue/test-utils';
import Div from '@/elements/Div/Div.vue';

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

it('div renders', () => {
  wrapper = mount(Div);

  const div = wrapper.get('div');

  expect(div).not.toBeNull();
  expect(wrapper.element.outerHTML).toMatchSnapshot();
});

describe('div formates classes correctly', () => {
  afterEach(() => {
    wrapper.unmount();
  });

  it('default element name', () => {
    wrapper = mount(Div, {
      props: {},
    });

    const div = wrapper.get('div');

    expect(div).not.toBeNull();
    expect(div.classes()).toEqual(['div']);
  });

  it('custom element name', () => {
    wrapper = mount(Div, {
      props: { elem: 'custom' },
    });

    const div = wrapper.get('div');

    expect(div).not.toBeNull();
    expect(div.classes()).toEqual(['custom']);
  });

  it('custom block name', () => {
    wrapper = mount(Div, {
      props: { block: 'parent' },
    });

    const div = wrapper.get('div');

    expect(div).not.toBeNull();
    expect(div.classes()).toEqual(['parent-div']);
  });

  it('custom block-element name', () => {
    wrapper = mount(Div, {
      props: { block: 'parent', elem: 'custom' },
    });

    const div = wrapper.get('div');

    expect(div).not.toBeNull();
    expect(div.classes()).toEqual(['parent-custom']);
  });

  it('with mods', () => {
    wrapper = mount(Div, {
      props: {
        mods: {
          enabled: true,
          falsy: false,
          theme: 'dark',
        },
      },
    });

    const div = wrapper.get('div');

    expect(div).not.toBeNull();
    expect(div.classes()).toEqual([
      'div',
      'div__enabled',
      'div__theme_dark',
    ]);
  });

  it('with all bem props', () => {
    wrapper = mount(Div, {
      props: {
        block: 'parent',
        elem: 'custom',
        mods: {
          enabled: true,
          falsy: false,
          theme: 'dark',
        },
      },
    });

    const div = wrapper.get('div');

    expect(div).not.toBeNull();
    expect(div.classes()).toEqual([
      'parent-custom',
      'parent-custom__enabled',
      'parent-custom__theme_dark',
    ]);
  });
});

it('slots', () => {
  wrapper = mount(Div, {
    slots: {
      default: '<button class="test-button" />',
    },
  });

  expect(wrapper.get('.test-button')).not.toBeNull();
  expect(wrapper.html()).toBe('<div class="div"><button class="test-button"></button></div>');
});
