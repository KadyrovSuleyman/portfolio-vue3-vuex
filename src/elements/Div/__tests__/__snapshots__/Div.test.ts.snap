// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`div renders 1`] = `<div class="div"></div>`;
